src/main/java/com/lorepo/icplayer/client/module/ordering/OrderingPrintable.java

```diff
@@ -21,63 +21,160 @@ public OrderingPrintable(OrderingModule model, PrintableController controller) {
 	}
 	
 	public String getPrintableHTML(String className, boolean showAnswers) {
-		String result = "<div class=\"printable_ic_ordering\" id=\"" + model.getId() +"\">";
-		
+		if (model.getPrintable() == PrintableMode.NO)
+			return null;
+
+		HashMap<String, String> printableState = model.getPrintableState();
+		String printableRepresentation = "";
+
+		if (printableState != null && printableState.containsKey("order"))
+			printableRepresentation = createPrintableRepresentationIfSetState(showAnswers, printableState);
+		else
+			printableRepresentation = createPrintableRepresentationIfNotSetState(showAnswers);
+		return PrintableContentParser.addClassToPrintableModule(
+				printableRepresentation, className, !model.isSplitInPrintBlocked());
+	}
+
+	private String createPrintableRepresentationIfSetState(
+			boolean checkAnswers, HashMap<String, String> printableState) {
+		String[] parsedItems = new String[model.getItemCount()];
+		List<String> unorderedItems = new ArrayList<String>();
+
+		String[] itemsIndexesFromState = getItemsIndexesInOrderFromState(printableState);
+		boolean isCorrectOrder = true;
+
+		for (int i = 0; i < itemsIndexesFromState.length; i++) {
+			int itemIndexFromState = Integer.parseInt(itemsIndexesFromState[i]);
+			int position = i+1;
+			for (int j = 0; j < model.getItemCount(); j++) {
+				OrderingItem item = model.getItem(j);
+				if (item.getIndex() == itemIndexFromState) {
+					if (!item.isCorrect(position))

```

Comment:
Nie czytelniej byłoby bez ifa? 
isCorrectOrder = item.isCorrectOrder(position)

Reply:
Faktycznie. Dodam zmianę

Reply:
Przejrzałem całą funkcję i nie można byłoby zaproponowanej zmiany zastosować.
Celem dodatkowego ifa jest:
Jeżeli choć jeden element nie jest Correct to ustaw na false.
Zaproponowana zmiana pozwoliłaby na ponowne ustawienie na true.

Reply:
Zrobiłem bez ifa.
Twoja propozycja zmieniała strukturę więc zmodyfikowałem ją na odpowiednią.